include/deal.II/matrix_free/operators.h

```diff
@@ -160,6 +160,12 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
+     * Get read access to the MatrixFree object stored with this operator.
+     */
+    const std_cxx11::shared_ptr<const MatrixFree<dim,Number> > &
```

Comment:
since initizalization of matrix-free operators require a shared pointer, i think it makes more sense to return this here instead of a reference to the object itself.